lib/game_team.rb

```diff
@@ -100,79 +100,37 @@ def self.get_goal_shots_by_game_team(game_teams)
     results = {}
     #each game, iterating through and adding the team_id as the key and
     #getting the shots and goals count
-    game_teams.each do |game_team|
-      results[game_team.team_id] ||= {"shots"=>0,"goals"=>0}
-      results[game_team.team_id]["shots"] += game_team.shots
-      results[game_team.team_id]["goals"] += game_team.goals
+    grouped_season = game_teams.group_by{|group|group.team_id}
+    grouped_season.each do |team_id, game_team|
+      results[team_id] ||= {"shots"=>0,"goals"=>0}
+      results[team_id]["shots"] = game_team.sum(&:shots)
+      results[team_id]["goals"] = game_team.sum(&:goals)
     end
```

Comment:
what does (&:shots) do? I know its a shortcut, but that's about it

Reply:
Since game_team is an array of game_teams, we would have to iterate through each game_team to get the game_team.shots. So what this does is skip that and grabs only the category :shots from the game_team array and sums them up

Reply:
actually naming convention should be the opposite so i'll change that to game_teams within the block iterator

Reply:
thats awesome

Reply:
cool, good to know